src/appleseed/foundation/curve/binarycurvefilewriter.cpp

```diff
@@ -0,0 +1,134 @@
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// Interface header.
+#include "binarycurvefilewriter.h"
+
+// appleseed.foundation headers.
+#include "foundation/core/exceptions/exceptionioerror.h"
+#include "foundation/math/vector.h"
+#include "foundation/curve/icurvewalker.h"
+#include "foundation/platform/types.h"
+
+// Standard headers.
+#include <cstring>
+
+using namespace std;
+
+namespace foundation
+{
+
+//
+// BinaryCurveFileWriter class implementation.
+//
+
+BinaryCurveFileWriter::BinaryCurveFileWriter(const string& filename)
+  : m_filename(filename)
+  , m_writer(m_file, 256 * 1024)
+{
+}
+
+void BinaryCurveFileWriter::write(const ICurveWalker& walker)
+{
+    if (!m_file.is_open())
+    {
+        m_file.open(
+            m_filename.c_str(),
+            BufferedFile::BinaryType,
+            BufferedFile::WriteMode);
+
+        if (!m_file.is_open())
+            throw ExceptionIOError();
+
+        write_signature();
+        write_version();
+    }
+
+    write_curves(walker);
+}
+
+void BinaryCurveFileWriter::write_signature()
+{
+    static const char Signature[11] = { 'B', 'I', 'N', 'A', 'R', 'Y', 'C', 'U', 'R', 'V', 'E' };
+
+    checked_write(m_file, Signature, sizeof(Signature));
+}
+
+void BinaryCurveFileWriter::write_version()
+{
+    const uint16 Version = 2;
+
+    checked_write(m_file, Version);
+}
+
+void BinaryCurveFileWriter::write_curves(const ICurveWalker& walker)
+{
+    write_basis(walker);
+    write_curve_count(walker);
+
+    int32 vertex_count = 0;
+
+    for (uint32 i = 0; i < walker.get_curve_count(); ++i)
+        write_vertex_properties(walker, i, vertex_count);
+}
+
+void BinaryCurveFileWriter::write_basis(const ICurveWalker& walker)
+{
+    const unsigned char basis = static_cast<unsigned char>(walker.get_basis());
+    checked_write(m_writer, basis);
+}
+
+void BinaryCurveFileWriter::write_curve_count(const ICurveWalker& walker)
+{
+    const uint32 curve_count = static_cast<uint32>(walker.get_curve_count());
+    checked_write(m_writer, curve_count);
+}
+
+void BinaryCurveFileWriter::write_vertex_properties(const ICurveWalker& walker,
```

Comment:
While technically true, this is probably better called `write_curve()` or something along these lines.

Reply:
Fixed.